src/components/Request/RequestControl.tsx

```diff
@@ -1,37 +1,69 @@
 import Button from "../Button";
+import { updateRequestStatus } from "../../backendCalls/requests";
+import { requestStatusType } from "../../dataModels/requests";
 
 const controlButtonStyle = {margin: "8px"};
-const messageRequiresBackend = () => {
-    alert(
-        "This feature is not implemented yet " +
-        "because it requires a persistent backend."
-    );
+
+interface requestControlProps {
+    requestId: string;
+    requestStatus: requestStatusType;
 };
 
-const RequestControl = (props: { datasetId: string }) => (
+
+const RequestControl = (props: requestControlProps) => (
     <div className="w3-panel">
         <span className="w3-text-green w3-large">Request Control:</span>
-        <p>
-            Approve or decline the request.
-        </p>
-        <div className="w3-panel w3-bar">
-            <Button
-                style={controlButtonStyle}
-                icon="fas fa-check"
-                color="green"
-                onClick={messageRequiresBackend}
-            >
-                approve
-            </Button>
-            <Button
-                style={controlButtonStyle}
-                icon="fas fa-times"
-                color="red"
-                onClick={messageRequiresBackend}
-            >
-                decline
-            </Button>
-        </div>
+        { props.requestStatus === "pending" ? (
+            <div>
+                <p>
+                    Approve or reject the request.
+                </p>
+                <div className="w3-panel w3-bar">
+                    <Button
+                        style={controlButtonStyle}
+                        icon="fas fa-check"
+                        color="green"
+                        onClick={() => updateRequestStatus(props.requestId, "approved")}
+                    >
+                        approve
+                    </Button>
+                    <Button
+                        style={controlButtonStyle}
+                        icon="fas fa-times"
+                        color="red"
+                        onClick={() => updateRequestStatus(props.requestId, "rejected")}
+                    >
+                        reject
+                    </Button>
+                </div>
+            </div>
+        ) : (
+            props.requestStatus === "approved" ? (
+                <div>
+                    <p>
+                        The request has been approved.
+                        You may withdraw it.
```

Comment:
The requester might want to make the required changes and then re-apply for an access to a rejected request, but once a request has been approved, should the Data steward have the right to reject it? In that case the requester must be provided reasons. I would like to open this for discussion since I am not exactly sure about the workflow here.

My idea is:
Rejected requests can be re applied.
Approved requests cannot be rejected or can only be rejected if there's a compelling reason.

Reply:
Thanks Mani, here we are getting already deep into production discussions which are beyond the scope of the sandbox.

We can discuss that at a later point. Do you have a place where we can keep track of these topics?

But in general, I think, the Data Steward always needs the right to withdraw the access at any point. That is our right as data controller and there are multiple occasion when a preliminary withdrawal of the access permissions will be required (e.g. the data subjects withdraws the consent).

When you would like to re-apply for a dataset, in my head, that is a new request. Not sure whether there is a lot of value of re-opening the request at a later point. But again that should be discussed later.

Reply:
Ok